pgmpy/inference/ExactInference.py

```diff
@@ -1274,8 +1274,15 @@ def query(self, variables, evidence):
         ----------
         variables: list
             list of variables for which you want to compute the posterior
-        evidence: dict
+        evidence: dict (default: {})
             a dict key, value pair as {var: state_of_var_observed}
+        virtual_evidence: dict (default: {})
+            a dict key, virtual message list pair as {var: [virtual_message_array]}
+            Note: in message-parsing BP, handling virtual evidence is more straightforward that in the junction
+            tree BP implemented in the `BeliefPropagation` class. Each virtual evidence becomes a virtual message
+            that gets added to the list of computed messages incoming to the variable node. There's no need to use
+            the `Inference._virtual_evidence()` method. Hence, a dict with arrays makes the code faster/cleaner 
+            than a list of TabularCPD.
```

Comment:
Comment on choice for virtual evidence implementation. Another option to be consistent with the main BeliefPropagation class is to let `query` take in `virtual_evidence` as a list of tabular CPD, transform it into a dict of var, np.array which is easier to parse, and then propagate the `virtual_evidence` as a dict into the code. I went to for the lighter option. Let me know what you prefer.